controller/src/main/java/io/pravega/controller/store/stream/PravegaTablesStreamMetadataStore.java

```diff
@@ -73,19 +76,25 @@
         this.completedTxnGC = new ZKGarbageCollector(COMPLETED_TXN_GC_NAME, zkStoreHelper, this::gcCompletedTxn, gcPeriod);
         this.completedTxnGC.startAsync();
         this.completedTxnGC.awaitRunning();
+        this.completedTxnGCRef = new AtomicReference<>(completedTxnGC);
         this.counter = new ZkInt96Counter(zkStoreHelper);
         this.storeHelper = new PravegaTablesStoreHelper(segmentHelper, authHelper, executor);
         this.executor = executor;
     }
 
-    private CompletableFuture<Void> gcCompletedTxn() {
+    @VisibleForTesting 
+    CompletableFuture<Void> gcCompletedTxn() {
         List<String> batches = new ArrayList<>();
         return withCompletion(storeHelper.expectingDataNotFound(storeHelper.getAllKeys(COMPLETED_TRANSACTIONS_BATCHES_TABLE)
                                                          .collectRemaining(batches::add)
                                                          .thenApply(v -> {
                                                                      // retain latest two and delete remainder.
                                                                      if (batches.size() > 2) {
-                                                                         return batches.subList(0, batches.size() - 2);
+                                                                         List<String> list = batches
+                                                                                 .stream().sorted(Comparator.comparingLong(Long::parseLong))
```

Comment:
This is a very inefficient sort. On every comparison you invoke `Long.parseInt`. Please convert the list to Long and then sort it.

Reply:
And since you only need the top 2, there may be more efficient methods that can do this in one pass, without sorting.

Reply:
Actually the list was going to be very small - typically 3 elements and this code was exercised once every day. so it would not have mattered so much. 
But i have changed it to use a one pass algorithm to exclude two largest numbers in the list.